### DISCUSSION TITLE
Client returns 403 for plain auth

### DISCUSSION THREAD

**Discussion by PeaceRebel** [rabbitmq/amqp091-go#194] [Q&A]
### Describe the bug

We're trying to connect to a rmq server using plain auth for which the client throws username or password not allowed. This is on go version 1.20 and amqp091-go v1.8.0. We have another version of our app using go 1.14 and streadway/amqp v1.0.0 on which the same url works fine.

We inspected the traffic with the help of wireshark and the server responds with Connection.Start but the client closes the connection for some reason.

### Reproduction steps

1. Connect to rmq server with amqp url

### Expected behavior

Client should not close the connection if server responded ok.

**Comment by Zerpet** [comment]
Basic authentication works fine, as shown in our tests. I think nobody will use this library otherwise.

If you can reliably reproduce this problem, with a new instance of RabbitMQ (e.g. start a new Docker container), and provide the code + repro steps, I will be glad to investigate.

**Comment by PeaceRebel** [comment]
This was a problem in one of our wrapper implementations.

**Reply by lukebakken** [reply]
> This was a problem in one of our wrapper implementations.

Thanks for letting us know.

In the future, when a very basic, necessary software feature is failing (like authentication), the most likely cause of the issue is _your code_. A lot of people jump to conclusions that errors are due to bugs in RabbitMQ or RabbitMQ client libraries, but as @Zerpet said, if authentication failed all the time nobody would use our software 😉
